src/handlers/evm/index.ts

```diff
@@ -122,7 +123,11 @@ export class EvmHandler extends Utils {
       params,
     });
 
-    if (params.fromIsNative) {
+    const skipAllowanceCheck =
+      params.fromIsNative ||
+      data.route.transactionRequest?.type === SquidDataType.DepositAddressCalldata;
```

Comment:
Might be worth adding a comment that it means it will be direct transfer instead of swap with contract, mentioning coral v2 ?